server/server.go

```diff
@@ -113,6 +113,33 @@ func Start(ctx context.Context, logger *zap.Logger, cfg *Config) error {
 	return http.ListenAndServe(cfg.Bind, nil)
 }
 
+func authHandler(middleware *samlsp.Middleware, handler http.Handler, initiateSessionPath string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		session, err := middleware.Session.GetSession(r)
+		if session != nil {
+			r = r.WithContext(samlsp.ContextWithSession(r.Context(), session))
+			handler.ServeHTTP(w, r)
+			return
+		}
+		if err == samlsp.ErrNoSession {
+			if initiateSessionPath == "" {
+				middleware.HandleStartAuthFlow(w, r)
+				return
+			}
+
+			if r.URL.Path == initiateSessionPath {
+				middleware.HandleStartAuthFlow(w, r)
+				return
+			}
+
+			handler.ServeHTTP(w, r)
+			return
+		}
+
+		middleware.OnError(w, r, err)
+	})
+}
```

Comment:
It would be nice if this wasn't mostly a copy-modify of `sampsp.Middleware#RequireAccount`. I'm going to explore idea by optionally wrapping the session provider with an "anonymous session" aware one...